editor/src/components/canvas/canvas-component-entry.tsx

```diff
@@ -93,26 +94,74 @@ const CanvasComponentEntryInner = React.memo((props: CanvasComponentEntryProps)
         }}
       >
         {canvasProps == null ? null : (
-          <CanvasErrorBoundary
-            filePath={canvasProps.uiFilePath}
-            projectContents={canvasProps.projectContents}
-            reportError={onRuntimeError}
-            requireFn={canvasProps.curriedRequireFn}
-            key={`canvas-error-boundary-${canvasProps.mountCount}`}
-          >
-            <RemoteDependencyBoundary
-              projectContents={canvasProps.projectContents}
-              requireFn={canvasProps.curriedRequireFn}
-            >
-              <DomWalkerWrapper {...canvasProps} clearErrors={localClearRuntimeErrors} />
-            </RemoteDependencyBoundary>
-          </CanvasErrorBoundary>
+          <CanvasInner
+            canvasProps={canvasProps}
+            onRuntimeError={onRuntimeError}
+            localClearRuntimeErrors={localClearRuntimeErrors}
+          />
         )}
       </div>
     </>
   )
 })
 
+function CanvasInner({
```

Comment:
extracting it to a component so that we can be sure that `canvasProps` are defined, since the `useInvalidatedCanvasRemount` hook also assigns the current count values to the `ref`s, and when they are undefined it creates a false positive on the next render (which breaks a lot of tests)

Reply:
should this be put inside a `React.memo`? honest question, I don't know if it would create issues with the refs

Reply:
Since it's an extracted inline component, I think that now without a `memo` it works exactly as the same before (meaning it will render at the same times when the online component was rendered, not more or less).
There might be a performance gain to adding `memo`, but this needs to be checked, and is unrelated to the extraction (again - since now it works exactly as before, and the parent component is `memo`ed)